java/dataset/src/main/cpp/jni_wrapper.cc

```diff
@@ -154,6 +155,19 @@ arrow::Result<std::shared_ptr<arrow::dataset::FileFormat>> GetFileFormat(
   }
 }
 
+arrow::Result<std::shared_ptr<arrow::dataset::FragmentScanOptions>>
+GetFragmentScanOptions(jint file_format_id,
+                       const std::unordered_map<std::string, std::string>& configs) {
+  switch (file_format_id) {
+#ifdef ARROW_CSV
+    case 3:
+      return arrow::dataset::CsvFragmentScanOptions::From(configs);
+#endif
+    default:
+      return arrow::Status::Invalid("Illegal file format id: " ,file_format_id);
```

Comment:
nit: I think we don't have proper C++ linting @lidavidm ?
```suggestion
      return arrow::Status::Invalid("Illegal file format id: ", file_format_id);
```